src/main/java/heronarts/lx/midi/surface/APC40Mk2.java

```diff
@@ -1105,14 +1108,15 @@ private void noteReceived(MidiNote note, boolean on) {
           } else {
             LXClip clip = channel.getClip(index);
             if (clip == null) {
-              clip = channel.addClip(index);
+              channel.addClip(index, this.shiftOn);
+            } else if (this.shiftOn) {
+              clip.loop.toggle();
+              sendClip(channelIndex, channel, index, clip);
```

Comment:
You'll be able to drop this line when we add the listener to clip.loop - the parameter listener will get the clip redrawn properly.